### ISSUE TITLE
Automatically switch to "select" mode when text is selected?

### ISSUE THREAD

**Issue by digitalmaster** [DCsunset/vscode-modal-editor#14]
Is it possible to automatically enable select mode when text is selected? This is a pretty powerful behavior i'm missing from ModalEdit since it enables me to have a different set of bindings for when text is selected.

Currently the only way seems to be to manually call `modalEditor.setSelectMode` after select actions. But this will still miss the case when I manually select text by mouse (don't judge 🙈 ).

Is there a way to do this?

**Comment by DCsunset** [comment]
I'm not sure if the event can be listened on in the editor, but there is an event called `onDidChangeTextEditorSelection` in the VSCode API. If you have some time, could you double check whether you can register any function for it?

If it's not doable outside the extension, I think we may consider allowing user to register their own function through this extension.

**Comment by digitalmaster** [comment]
>  If you have some time, could you double check whether you can register any function for it?

Yup, looks like you can. In fact, [looks like](https://github.com/DCsunset/vscode-modal-editor/blob/c3a4469fd8fcc788ffb112e9ae96b168606ee332/src/extension.ts#L18) you already have a function registered (tho it's not clear to me exactly what it does 🙈).

I just tested it and it does work - tho seems to trigger when i'm just moving around for some reason 🤔

**Comment by DCsunset** [comment]
Currently, the function implements inclusive range. I was actually asking if you can register a function in VSCode settings without modifying the code.

Probably you can't do that. In that case, I think it's a good idea to support user-defined event handlers.

**Comment by digitalmaster** [comment]
I love the idea of being able register commands to run on editor events. Something like:
```js
{
  "events": {
    "onTextSelect": "modalEditor.setSelectMode",
    "onSearch": "closeFindWidget"
  }
}
```

**Comment by digitalmaster** [comment]
Sorry for spamming here.. but found another potential use case for this 🙈 

I'm looking for a way to implement [this behavior](https://github.com/johtela/vscode-modaledit/blob/2c83f43c9fcff646622354e8db5c4f06f1d43adf/src/commands.ts#L304) (tldr: clearing selection when entering normal mode). 

If we could bind events the the mode change action I could also achieve the same thing :)

**Comment by DCsunset** [comment]
You can easily achieve it without event by calling `clearSelections` after using `setNormalMode` since you can bind a key to a list of commands.

In the mean time, I think it's a better idea to go through README thoroughly before you add a lot of settings, because It can help you understand the extension better and possibly save your some time.

**Comment by digitalmaster** [comment]
Oh interesting. Can you say more about this. I actually have gone through the entire readme a couple of times 🙈 tho gotto admit that it's not immediately obvious to me how we'd bind the `clearSelections` command to an event not triggered by a config binding  🤔 Isn't the setNormalMode command bound to a vscode esc key binding? Gonna take another look...

**Comment by DCsunset** [comment]
You can overwrite the default keybinding for `Esc` to `modalEditor.executeCommand`, similar to the code in section https://github.com/DCsunset/vscode-modal-editor?tab=readme-ov-file#command-context

**Comment by digitalmaster** [comment]
Yup, that did the trick. Thanks for the help. Apologies for the confusion 🙏 

For reference, 
```json
  {
    "key": "escape",
    "command": "modalEditor.executeCommand",
    "when": "editorTextFocus",
    "args": [
      "modalEditor.setNormalMode",
      "modalEditor.clearSelections",
    ]
  },

```
